### ISSUE TITLE
questions about dataset and model structure

### ISSUE THREAD

**Issue by fenneccat** [lpworld/PURS#7]
1. What does click (or rating) means? In the test.txt, the value varies from 0 to 10. However, in the paper, when testing MovieLens model, ratings are binarized with threshold 3.5. Which one is right? Does rating should be ranged between 0 and 1, or not?

2. In figure 3, it seems there are two models: Base, and Unexpected. Each model predicts r_ui (Click Through Rate) and Unexp_Factor_ui respectively. However, in code, it seems they are actually one model and jointly trained with logits `item_b + concat + user_b + unexp_factor*unexp`. The label to train for this entire logits is binary value (click or non-click value) not CTR. Am I understood correctly? 

3. According to paper, r_ui seems to calculated by pass through MLP layer with three inputs: user embedding, long preference (history), and item embedding. However, in the code, final logits is like below
self.logits = item_b + concat + user_b + unexp_factor*unexp 
It seems `item_b + concat + user_b` part is r_ui.
Why does bias item_b and user_b is added at last? And why concat only considers long preference (history) and item embedding? 
(model.py:line.49: concat = tf.concat([long_preference, item_emb], axis=1))

**Comment by lpworld** [comment]
Hey, thanks for your interest in our paper.

(1) Indeed, the ratings should be binarized. The provided dataset is a snapshot of the Alibaba-Youku dataset (per requested by one of the readers who wants to know more about the data structure) And the "click" represents the number of clicks in the user records. You can binarize it by click = (click==0).

(2) Yes, there is only one model, which contains two "components", one for the relevance objective and the other for the unexpectedness objective.

(3) Adding the bias terms item_b and user_b is an engineering practice that we found out to be useful when we launch the model in the company. You can also include the short-preference, but that actually does not improve the recommendation performance.

**Comment by fenneccat** [comment]
Thank you for the great reply!
